### DISCUSSION TITLE
Welcome to Yarle Discussions!

### DISCUSSION THREAD

**Discussion by akosbalasko** [akosbalasko/yarle#118] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gitophile** [comment]
Great work.  Want to be able to keep up with changes it looks like are made regularly.  So....  Pardon my noobie q but is it best to **_clone_** and manually do this or some other way better.  Plan to do a custom template too.  

Thanks Greg

**Reply by akosbalasko** [reply]
Hi @gitophile ! 
Thank you!
Please raise your ideas in the Issues tab just to avoid double-work and there we can agree on who can or feel willing to implement that. If you would like to do it, then yes, the common way is to clone the repo and submit a PR on your implementation.

**Comment by jalurgemilang** [comment]
I am contemplating moving out of Evernote into Obsidian. Your tool is a savior to me because you saved the Created Date and Modified Date into the Windows Explorer file itself, so all my notes are not shown will today's date in Windows Explorer. I don't think there's any other conversion tool that has done this. 

However, I notice that the filename does have the date and time. For example, my Hello World note will be exported as "20210630111111 Hello World.md" file. When I use Obsidian to open the files, I will see a lot of numbers (timestamp) on the left panel of Obsidian instead of the actual note's title. How can I make Yarle export without the timestamp in the file name?

**Reply by akosbalasko** [reply]
Hi @jalurgemilang !
I'm glad you like the tool! :)
The timestamp in the filename is the Zettekasten Id, if you would like to do not generate them, please set 'isZettelkastenNeeded' to false ( meaning that untick its checkbox in the config page)

cheers, 
akos

**Reply by jalurgemilang** [reply]
Thank you. That must be something I missed that setting out. I appreciate you taking the time to respond.

Another feedback, I notice that some of my notes created date and modified date is not being exported out consistently. I am referring to the Properties of the file, not the content inside the markdown. This is using Windows OS.
I have to use Yarle in MacOS on the same Evernote notebook did not observe the inconsistency. I wonder if it was a bug with Windows. Thank you.

**Comment by eitang** [comment]
Looking to run away form Evernote into Obsidian... 🏃🏽‍♂️
This tool is the closet I've found to get something useful, thank you so much for building and supporting it! 🙏🏼
However the highlighting export doesn't seem to work. 
I read that it's supported and even saw it in the `outputFormat` configuration.

However the highlighting doesn't come out on the .md file, tested this across multiple notes.
See an examples below 👇🏼

**ENEX file formatting**
`<div><b><span style="--en-highlight:red;background-color: #fec1d0;">My ego demands—for myself—the success of my team. —BILL RUSSELL</span></b></div>`


**MD file output**
```
**My ego demands—for myself—the success of my team. —BILL RUSSELL**

```

Used yarle-evernote-to-md.4.0.9.Setup.exe 

GitHub doesn't let me to upload .enex files, so I'm attaching my original file ZIPed in case you want to see it.
 
[Kindle highlights from The Captain Class A New Theory of Leadership.enex.zip](https://github.com/akosbalasko/yarle/files/6681286/Kindle.highlights.from.The.Captain.Class.A.New.Theory.of.Leadership.enex.zip)


Thanks again for creating this!

**Reply by akosbalasko** [reply]
Hi @eitang !

I'm glad you like it! :) 

Thank you for raising this up, it looks like that the enex note specifies highlights differently than yarle tries to identify it: 
- in Yarle a highlight is a span element with a argument named '-evernote-highlight=true'
- in your enex file it is a span with an argument named 'en-highlight' .
therefore Yarle cannot identify the highlights.

In your enex file i can see that it is exported via EN v10. 
 Currently EN 7xx is supported only, EN 10 is still not, because it is a huge improvement as the data structure of the enex file in EN 10 has changed significantly.  But it's on the roadmap definitely.
So, could you please use the legacy Evernote (7xx) to export your notes? It allows you btw to export everything without limitation of the amount of the noted including large ones as well.

Please feel free to contact if you still face issues in the tool.

Thank you!

**Reply by eitang** [reply]
Thanks for the quick reply!

Tried using Evernote's legacy version 7.14 on Mac and 6.25 on Windows, the highlighting export still doesn't work :-( 

Attaching the enex export files here just in case they're helpful.
I might run a find replace on all of the highlights and see if it works.
Can you pls send me a full example of what a highlight that Yarle recognizes looks like? 

If it works, I'll share my findings :-) 

[Archive.zip](https://github.com/akosbalasko/yarle/files/6682095/Archive.zip)

Thanks! 🙏🏼

**Reply by akosbalasko** [reply]
Hi @eitang ,
ohh, okay I'm sorry for hearing this. Let me investigate it further and provide a fix within a couple of days.

I'll keep you posted.

Thanks for reporting this!

**Reply by eitang** [reply]
No worries! Thank you!

I think I found a way to solve it, but I don't know regex, so need to learn a bit...
What I did to validate that my solution can work is to replace: 
--en-highlight:red;

With
-evernote-highlight:true

Did a simple find replace with Sublime, now I'm learning Regex to find all the highlights, no matter the color...
If you know regex it'll help :-)

Basically instead of red, find all 
--en-highlight:red;

**Reply by akosbalasko** [reply]
yeah, I found the same. If you have some time for me, i'll release the fixed version possibly tomorrow. Today I'm not sure, because i'm on holiday with the family. ;)

**Reply by eitang** [reply]
Of course I've got time.
Thank you so much! Really appreciate it!

**Reply by akosbalasko** [reply]
hi @eitang ,
I released a version with the fix (4.0.10), feel free to try it out, any feedbacks and further improvement ideas are welcome! 

cheers!

**Reply by eitang** [reply]
Thank you so much!
Seems to be working :-) 

Will share ideas if I'll think of anything.

**Reply by eitang** [reply]
Found one for you :-)

Tags are imported as code they're surrounded by ```.
Example:
```
#tags
```

As a result they aren't imported as tags, but as code text.

**Comment by yukonquest** [comment]
Good Day!  I am a brand new Obsidian User, I found YARLE in the Obsidian Forums.  I used the desktop Windows version, I have Windows 8.1. All is working well, thank you so much for putting this together.

1.  My Evernote export has 25 enex files (I have a lot of notes, 12 years worth). Each enex file takes 4-6 hours to complete. I will be at this for days. Is there a way to pickup all the enex files in succession to avoid having to do them individually?
2.  The instructions state that after the conversion, to simply “drag” the converted files into Obsidian. The structure that has been created is: /notes//Evernote [2] (where the converted notes are) & /notes/Evernote [2]//_resources (where the attachments are). I tried dragging in ‘all’ the files under Evernote [2], without success. Can you elaborate on how to complete the import process?

Thank you in advance for you assistance. Once I get this all working, this will be a huge timesaver for me (even if it takes days).
On Obsidian Forum:  yukonquestm@gmail.com
Michael

**Reply by akosbalasko** [reply]
Hi @yukonquest !

To answer your questions, nope, currently Yarle has no parallelization implemented unfortunately, the enex files are converted one by one. But it sounds like a great improvement for the future, so I add it as a feature request on the issues page. Thank you! 

On the second I just wanted to express that just simply move the converted MD files into the folder of the Obsidian vault you use. It can be done in the file system by hitting F6 in Midnight commander or typing ```mv``` in terminal,  and let Obsidian rebuild the index during its start-time. 

Welcome to Obsidian, have a wonderful time here!

**Comment by fair21comic** [comment]
Hi @akosbalasko 

Like many others, I am migrating from Evernote to Obsidian and I find your tool to be _almost_ perfect for the job, so thank you so much for developing it.

My method has been to save the ENEX files from Evernote (you can do 100 notes at a time, and I only have ~300 notes) and then use YARLE to convert to Obsidian format.  However I notice that while your tool does preserve the created/modified date somewhere in that it appears correctly at the top of each note:

![image](https://github.com/user-attachments/assets/431efe4b-1ab7-47b2-bb40-2db1ec03fdff)

Obsidian seems to use the date of conversion to sort the notes.  Indeed, if I hover over a note, all of them have the date of conversion, which is the same for each note, so I end up with just alphabetical ordering:

![image](https://github.com/user-attachments/assets/d597d04d-0c6b-4854-a43a-b2bd7b61c783)

I've had a look round both in the settings for YARLE and online forums and haven't come across this particular problem - do you have any ideas?

Jamie

**Reply by TjenWellens** [reply]
There is a difference between the date in properties (aka frontmatter) and the date in of the file itself.

A file on your OS (windows/linux/mac) has some dates hidden in the file.
I'm guessing that is your second screenshot: obsidian reading the file's created/modified date.
This should match if you use file explorer (or your OS's equivalent, not obsidian)
<img width="389" alt="Screenshot 2024-09-03 at 13 42 03" src="https://github.com/user-attachments/assets/a85e2ab1-a23b-4827-981d-3a04c35a345e">

Your first screenshot is your file's frontmatter, and if in obsidian you click 'source mode' (or open the readme file in another app) you will see that the file literally has text at the beginning with those properties
<img width="635" alt="image" src="https://github.com/user-attachments/assets/d5d42c99-6f56-49c9-860c-d4e81c5be35c">

These two places where dates are stored, are completely separate.
And they won't even be in sync in the future, with the frontmatter 'last updated at' never changing, even though the file's updated date changes.
(obsidian has plugins for that)

**Reply by TjenWellens** [reply]
yarle should make have both dates the same, because it uses the utimes library to do exactly that.

Maybe you are losing dates them when you're copying the files.
How those 'hidden in file' dates are copied/lost depends on your OS (windows/linux/mac).
I remember having similar problems on my mac.

mac:
> the built in cp command actually handles it
>  -p    Cause cp to preserve the following attributes of each source file in the copy: modification time, access time [...]
src: https://apple.stackexchange.com/questions/80485/can-timestamps-be-preserved-when-copying-files-on-os-x

Windows:
> When you move a file to a different folder then its creation date will not change.
> When you copy a file do a different folder then its creation date will change.
src: https://superuser.com/questions/1748898/how-to-copy-file-folder-and-preserve-their-creation-date-on-windows-10

linux:
> Note that most Linux filesystems don't store dates of creation — the best you can have is date of last modification
src: https://askubuntu.com/questions/1085231/how-can-i-copy-a-folder-without-altering-their-original-dates-of-creation

Check with your 'file explorer'/'finder' (or your OS equivalent) first the date of the original output of yarle.
And compare this wherever you copied it to, and see if the dates changed because of copying.

**Comment by fair21comic** [comment]
Hi there

Thanks very much for getting back to me - sorry about my delayed reply; I
have been out of the country.

I think the issue is at the stage of ENEX import into Yarle.  The way I got
my note sinto ENEX format was by doing it manually in EverNote; it allows
you to do 100 notes at a time, and because I have 297 notes, I ended up
with three ENEX files.

Obviously, because these are new files, the ENEX files all have today's
date as a creation date.

I don't move or copy these files before importing them to YARLE, so I don't
think it is a move/copy issue.

Maybe I need to convert my notes into ENEX files in a different way?
Perhaps it needs to be one per note?

What would you suggest?

Many thanks again

Jamie

**Reply by TjenWellens** [reply]
Hey

Your screenshot shows the date of 2014 showing up correctly in the frontmatter (aka "properties", aka top of the markdown content)
This means yarle is reading the date correctly.

The moving of the enex files does not matter, because yarle looks inside the enex files.
Enex files are simple xml files, you can open them in a standard text editor and see the contents.
There will be a `<note>...</note>` entry per note (so about a 100 per enex file in your case).
Example:
```
<note>
    <title>Conference: SocratesBE 2022</title>
    <created>20220710T120501Z</created>
    <updated>20231109T160405Z</updated>
    <tag>published</tag>
    <tag>i.conference</tag>
    <tag>conference.socratesbe22</tag>
    <note-attributes>
      <author>Tjen Wellens</author>
      <source>desktop.mac</source>
    </note-attributes>
    <content>
        ... // your actual formatted text content: headers, paragraphs, checkboxes, etc
    </content>
    <resource> // once per attachment(images, pdfs, etc) in the note
        <data encoding="base64">
            ... // attachment data is base64 encoded, so not human readable
        </data>
        <mime>image/jpeg</mime>
          <width>4032</width>
          <height>3024</height>
          <resource-attributes>
              <file-name>Image from iOS.jpg</file-name>
              <source-url>...</source-url>
          </resource-attributes>
    </resource>
    <resource>...</resource> 
    <resource>...</resource>
</note>
```

oversimplified and ignoring attachments:
1. Yarle reads each `<note>` in the .enex file, and creates a file for it. (file's creation-date and updated-at date = today/now)
2. It reads the `<created>`, `<updated>` and prints it as frontmatter at the top of the file.
3. It reads the `<content>` and prints is next in the file (after changing it from evernote-xml to markdown).
4. It reads the `<created>`, `<updated>` and changes the file's creation-date and updated-at date.

Your frontmatter has the correct date, so the problem should be at yarle-step-4 or later.
Either:
- A) step 4 failed
- or
- B) you lose the date by copying the .md files (eg `1918 flu pandemic.md`)

I myself had (B) happen to me.
When copying the .md files from the yarle export location to where I wanted them to be, I lost the file's creation-date and updated-at date (while the frontmatter stayed correct).

So it's not the copying of the enex files that can cause (B), but the copying of the exported .md files.

**Comment by fair21comic** [comment]
Hi there

Thanks again so much for getting back to me.

I seem to have solved my problem: I did what I did before but outputted the
YARLE output directly into the Obsidian Vault folder.  Notes are sorting by
the dates in the note headers now.

Thanks so much for your help and for creating this great tool!

Jamie
